packages/calcite-components/src/components/menu-item/menu-item.tsx

```diff
@@ -266,61 +266,54 @@ export class CalciteMenuItem implements LoadableComponent, T9nComponent, Localiz
   };
 
   private keyDownHandler = async (event: KeyboardEvent): Promise<void> => {
-    // opening and closing of submenu is handled here. Any other functionality is bubbled to parent menu.
-    if (event.key === " " || event.key === "Enter") {
-      this.selectMenuItem(event);
-      if (
-        this.hasSubmenu &&
-        (!this.href || (this.href && event.target === this.dropdownActionEl))
-      ) {
-        this.open = !this.open;
+    const { hasSubmenu, href, layout, open, submenuItems } = this;
+    const key = event.key;
+    const targetIsDropdown = event.target === this.dropdownActionEl;
+    if (key === " " || key === "Enter") {
+      if (hasSubmenu && (!href || (href && targetIsDropdown))) {
```

Comment:
I think 

```ts
!href || (href && targetIsDropdown)
```
can be simplified as 

```ts
!href || targetIsDropdown
```

Reply:
Refactored this section to store all re-used vars / checks.